### ISSUE TITLE
Clamav detects malicious content in compressed file but its not detecting anything in same uncompressed file.

### ISSUE THREAD

**Issue by deendhayal** [Cisco-Talos/clamav#568]
Clamav software detects malicious content in compressed file (zip) where in it is not detecting anything suspicious in same uncompressed file.

Malware it detected - Andr.Spyware.Mobtool-7152523-0 FOUND\u0000

ClamAV version used to scan the file : Version: 26526 , Sig : 1982031, f_level: 90

**Comment by micahsnyder** [comment]
Hi @deendhayal 

That's how this signature works.  It detects Android malware.  Android apps (APK files) are Zip files. 

The signature itself is:
`Andr.Spyware.Mobtool-7152523-0;Engine:51-255,Container:CL_TYPE_ZIP,Target:0;(0&1&2)|(3&4&5);537962657253746172;416c65782055646f6666;4e6169726f6269;536d617274486f6d65;4a6f686e20536d697468;4d6f73636f77`

which when decoded to a more human readable format can be read as:
```
VIRUS NAME: Andr.Spyware.Mobtool-7152523-0
TDB: Engine:51-255,Container:CL_TYPE_ZIP,Target:0
LOGICAL EXPRESSION: (0&1&2)|(3&4&5)
 * SUBSIG ID 0
 +-> OFFSET: ANY
 +-> SIGMOD: NONE
 +-> DECODED SUBSIGNATURE:
SyberStar
 * SUBSIG ID 1
 +-> OFFSET: ANY
 +-> SIGMOD: NONE
 +-> DECODED SUBSIGNATURE:
Alex Udoff
 * SUBSIG ID 2
 +-> OFFSET: ANY
 +-> SIGMOD: NONE
 +-> DECODED SUBSIGNATURE:
Nairobi
 * SUBSIG ID 3
 +-> OFFSET: ANY
 +-> SIGMOD: NONE
 +-> DECODED SUBSIGNATURE:
SmartHome
 * SUBSIG ID 4
 +-> OFFSET: ANY
 +-> SIGMOD: NONE
 +-> DECODED SUBSIGNATURE:
John Smith
 * SUBSIG ID 5
 +-> OFFSET: ANY
 +-> SIGMOD: NONE
 +-> DECODED SUBSIGNATURE:
Moscow
```

If you have good reason to believe that the file it detected is not malware, please submit a false positive report to https://www.clamav.net/reports/fp

Best,
Micah

**Comment by deendhayal** [comment]
But if we scan the same content in uncompressed file format , its not detecting any virus. How that is possible ? So is that something issue with sacnning zip format files ?

**Comment by micahsnyder** [comment]
The reason is that `Container:CL_TYPE_ZIP` option in the signature prevents it from alerting if the content isn't in a ZIP (or APK).

**Comment by deendhayal** [comment]
So , Is this the common behaviour with clamav ? But there were few other zip files scanned , clamav didn't detect this kind of malware in those files.
Is this signature varies based on clamav version ?

**Comment by micahsnyder** [comment]
This behavior is specific to this signature. This signature was written to detect APKs (which are ZIPs), so that's why it only alerts on the APK before you extract it and not after. It's not an APK anymore if you extract it. Most ClamAV signatures do not work this way.

If you have specific file(s) that you are worried about and want to use as an example, you can upload them to VirusTotal and share a link to the page on VirusTotal.

**Comment by deendhayal** [comment]
But we have't configured any specific signature while installing clamav software . How can we mitigate this issue ? our expectation is , clamav should detect virus in the same fashion whether it is a zip file or just a normal file. How we can acheive that ?

**Comment by micahsnyder** [comment]
The issue you described is not a bug as far as I can tell. 

Android packages are either APK archives or AAB (Android Application Bundle) archives.  Both of these are specialized ZIP archives which means they are ZIP compressed files containing specific filenames and directory names:
- APK: https://en.wikipedia.org/wiki/Apk_(file_format)
- AAB: https://developer.android.com/guide/app-bundle/app-bundle-format

The signature that you are concerned about, "Andr.Spyware.Mobtool-7152523-0", is designed to detect malicious Android software.  Specifically it is aiming to detect an APK, which means that it is a ZIP containing specific stuff. This is why it won't alert if you unzip it. If you unzip it, it is no longer an APK.

If I am misunderstanding the nature of the issue, please provide a copy of the file, or a link to the file on VirusTotal as I requested so that I can investigate further.

**Comment by SecsAndCyber** [comment]
TL;DR ClamAV Signatures detect files. The signature in question specifically targets zip archives, unzipping it means it is not a zip archive.

More specifically, your premise is wrong. It would be fantastic if all of our clamav signatures were so precisely targeted that the substrings referred to "malicious content" as you stated in the original bug report. That is not what we do when we are creating and reviewing signatures. Our goal is to create signatures that will alert on malicious files while attempting to not alert on innocuous ones.

An interesting corner case comes up when writing detection for high profile malicious scripts. Often if the signature merely alerts on the novel malicious content then it will misfire and also alert on security blog posts which have embedded the malicious content in the text of their article.

As such, overall, across all of the officially published clamav signature set, I would hazard a guess that the majority of subsig content does not refer to the actual "malicious content". 

Now to the signature being discussed here, `@micahsnyder` is absolutely correct. The malicious file is an Android package. The signature substrings are not themselves indicative of "malicious content", they merely appear together within files that are a part of the malicious package being detected. As such, their appearance in other contexts are not an indicator of `Andr.Spyware.Mobtool`.
